src/overlap_checker.cpp

```diff
@@ -177,81 +153,98 @@ are_bboxs_disjoint(const Bnd_OBB &b1, const Bnd_OBB& b2, double tolerance)
+	unsigned num_parallel_jobs = 4;
 	double
 		bbox_clearance = 0.5,
 		max_common_volume_ratio = 0.01;
 	std::vector<double> imprint_tolerances = {0.001, 0};
 
 	{
-		CLI::App app{"Perform imprinting of BREP shapes."};
-		app.add_option(
-			"input", path_in,
-			"Path of the input file")
-			->required()
-			->option_text("file.brep");
-		app.add_option(
-			"-j", num_parallel_jobs,
-			"Parallelise over N threads")
-			->option_text("N")
-			->default_val(4)
-			->check(CLI::Range(1, 1024));
-		app.add_option(
-			"--bbox-clearance", bbox_clearance,
-			"Bounding-boxes closer than C will be checked for overlaps")
-			->option_text("C");
-		app.add_option(
-			"--imprint-tolerance", imprint_tolerances,
-			"Faces, edges, and verticies will be merged when closer than T")
-			->option_text("T")
-			->expected(1, 10);
-		app.add_option(
-			"--max-common-volume-ratio", max_common_volume_ratio,
-			"Imprinted volume with ratio <R is considered acceptable")
-			->option_text("R")
-			->check(CLI::Range(0., 1.));
-		CLI11_PARSE(app, argc, argv);
-	}
+		const char * doc = "Perform imprinting of BREP shapes.";
+		const char * usage = "input.brep";
+
+
+		std::stringstream stream;
+
+		stream << "Parallelise over N[=" << num_parallel_jobs << "] threads";
+		auto help_num_par_jobs = stream.str();
+		stream = {};
+
+		stream << "Bounding-boxes closer than C[=" << bbox_clearance << "] will be checked for overlaps";
+		auto help_bbox_cl = stream.str();
+		stream = {};
+
+		stream
+			<< "Faces, edges, and verticies will be merged when closer than T[="
+			<< imprint_tolerances[0] << ']';
+		auto help_imp_tol = stream.str();
+		stream = {};
+
+		stream
+			<< "Imprinted volume with ratio <R[="
+			<< max_common_volume_ratio << "] is considered acceptable";
+		auto help_max_common = stream.str();
+		stream = {};
+
+		cxx_argp::parser argp(1);
+		argp.add_option(
+			{"jobs", 'j', "N", 0, help_num_par_jobs.c_str()}, num_parallel_jobs);
+		argp.add_option(
+			{"bbox-clearance", 1024, "C", 0, help_bbox_cl.c_str()}, bbox_clearance);
+		argp.add_option(
+			{"imprint-tolerance", 1025, "T", 0, help_imp_tol.c_str()}, imprint_tolerances);
+		argp.add_option(
+			{"max-common-volume-ratio", 1026, "R", 0, help_max_common.c_str()}, max_common_volume_ratio);
+
+		if (!argp.parse(argc, argv, usage, doc)) {
+			return 1;
+		}
 
-	// make sure parameters are sane!
-	{
+		const auto &args = argp.arguments();
+		assert(args.size() == 1);
+		path_in = args[0];
+
+		if (num_parallel_jobs < 1 || num_parallel_jobs > 1024) {
+			LOG(ERROR)
+				<< "Number of parallel jobs must be between 1 and 1024.\n";
+			return 1;
+		}
 		unsigned max_parallel = std::thread::hardware_concurrency();
-		if (max_parallel && num_parallel_jobs > max_parallel * 2) {
-			spdlog::warn(
-				"requesting significantly more than the number of cores ({} > {}) is unlikely to help",
-				num_parallel_jobs, max_parallel);
+		if (max_parallel && num_parallel_jobs > max_parallel * 2 + 4) {
+			LOG(WARNING)
```

Comment:
why?

Reply:
it's CPU bound code.  I didn't want to warn users if they picked a couple more threads (so the scheduler always has work to do) and wasn't sure if/how hyperthreading was going to picked up

Reply:
are you threading with OpenMP?

Reply:
just `pthreads` as it didn't seem complicated enough to warrant anything else

Reply:
ok, the reason I ask is openmp has a function that returns the number of cores, I guess pthreads has the same? SO suggests https://stackoverflow.com/questions/150355/programmatically-find-the-number-of-cores-on-a-machine

Reply:
yes, `max_parallel` is set from that standard `std::thread::hardware_concurrency` API.  note that it reports hyper-threads instead of just "cores" for me under Linux

I wanted to let the user specify larger values for the number of parallel jobs (which is especially useful for my testing) but thought it would be useful to fail when they start requesting silly big numbers